app/models/video.rb

```diff
@@ -2,4 +2,9 @@ class Video < ApplicationRecord
   has_many :user_videos, dependent: :destroy
   has_many :users, through: :user_videos
   belongs_to :tutorial
+  validates :title, presence: true
+
+  def self.by_tutorial
+    order(:position).group_by(&:tutorial)
```

Comment:
Groups the videos by tutorial. I couldn't figure out how to do this in Active Record but may come back to it if we have time.